.github/workflows/ci_python.yml

```diff
@@ -0,0 +1,175 @@
+            # We shouldn't need regular dependencies, since they come from
+            # the `pip install` of kaskada later.
+            poetry install --only test
+          run: |
+            set -e
+            pip3 install kaskada --find-links dist --force-reinstall
+            pytest
+
+  windows:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        target: [x64, x86]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          architecture: ${{ matrix.target }}
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          args: --release --out dist
+          sccache: 'true'
+          working-directory: ${{ github.workspace }}/python
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: ${{ github.workspace }}/python/dist
+      - name: pytest
+        if: ${{ !startsWith(matrix.target, 'aarch64') }}
+        shell: bash
+        run: |
+          set -e
+          pip install kaskada --find-links dist --force-reinstall
+          pip install poetry
+          cd ${{ github.workspace }}/python
+          # We shouldn't need regular dependencies, since they come from
+          # the `pip install` of kaskada above.
+          poetry install --only=test
+          pytest
+
+  macos:
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        target: [x86_64, aarch64]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          args: --release --out dist
+          sccache: 'true'
+          working-directory: ${{ github.workspace }}/python
+      - name: Upload wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: ${{ github.workspace }}/python/dist
+      - name: pytest
+        if: ${{ !startsWith(matrix.target, 'aarch64') }}
+        shell: bash
+        run: |
+          set -e
+          pip install kaskada --find-links dist --force-reinstall
+          pip install poetry
+          # We shouldn't need regular dependencies, since they come from
+          # the `pip install` of kaskada above.
+          poetry install --only-tset
+          pytest
+
+  sdist:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build sdist
+        uses: PyO3/maturin-action@v1
+        with:
+          command: sdist
+          args: --out dist
+          working-directory: ${{ github.workspace }}/python
+      - name: Upload sdist
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels
+          path: ${{ github.workspace }}/python/dist
+
+  release:
```

Comment:
The release downloads the wheel we produced on each OS and then publishes to PyPI using `maturin upload`.